core/invoker/src/main/scala/whisk/core/invoker/Invoker.scala

```diff
@@ -87,6 +89,45 @@ object Invoker {
 
     val msgProvider = SpiLoader.get[MessagingProvider]
     val producer = msgProvider.getProducer(config, ec)
+
+    val proposedInvokerId = if (args.length == 1) { args(0).toInt } else { -1 }
+    val invokerName = sys.env.getOrElse("INVOKER_ASSIGNED_NAME", (() => {
```

Comment:
you could also read this via whisk config - we've generally avoid reading anything from sys env in the past.
it would fail if you start up the invoker without an id.